modules/solid_properties/include/materials/SolidPropertiesMaterial.h

```diff
@@ -0,0 +1,80 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Material.h"
+#include "DerivativeMaterialInterface.h"
+
+/**
+ * Base class for defining solid property materials.
+ */
+class SolidPropertiesMaterial : public DerivativeMaterialInterface<Material>
+{
+public:
+  static InputParameters validParams();
+
+  SolidPropertiesMaterial(const InputParameters & parameters);
+
+  /**
+   * Solid name
+   * @return string representing solid name
+   */
+  virtual const std::string & solidName() const;
```

Comment:
Is this necessary?

Reply:
No, probably not. I just copied that from the fluid properties module, which has names. I'll remove it